src/System.Private.CoreLib/shared/System/Math.cs

```diff
@@ -538,17 +538,31 @@ public static decimal Max(decimal val1, decimal val2)
 
         public static double Max(double val1, double val2)
         {
-            if (val1 > val2)
+            // When val1 and val2 are both finite or infinite, return the larger
+            //  * We count +0.0 as larger than -0.0 to match MSVC
+            // When val1 or val2, but not both, are NaN return the opposite
+            //  * We return the opposite if either is NaN to match MSVC
+
+            if (double.IsNaN(val1))
             {
-                return val1;
+                return val2;
             }
 
-            if (double.IsNaN(val1))
+            if (double.IsNaN(val2))
             {
                 return val1;
             }
 
-            return val2;
+            // We do this comparison first and separately to handle the -0.0 to +0.0 comparision
```

Comment:
Just as a note here: This optimization is completely fine for the JIT to do since the IEEE spec indicates that comparisons shall ignore the sign of zero and for `Min`/`Max`, the JIT specifies that either x or y can be returned if `x == y`.

I opted to respect the sign of zero here since the MSVC implementation does and since Annex F of the C Language specification (which dictates IEC 60559 compliance) has a footnote indicating that respecting the sign of zero here would be "ideal".